### PR TITLE
Admin orders: Closes #42

### LINKED ISSUE #42: 21: Admin Orders
As an Admin 
When I visit the dashboard 
Then I can see a listing of all orders 
And I can see the total number of orders for each status ("Ordered", "Paid", "Cancelled", "Completed") 
And I can see a link for each individual order 
And I can filter orders to display by each status type ("Ordered", "Paid", "Cancelled", "Completed") 
And I have links to transition between statuses

I can click on "cancel" on individual orders which are "paid" or "ordered" 
I can click on "mark as paid" on orders that are "ordered"
I can click on "mark as completed" on orders that are "paid"

### PR DESCRIPTION
**Pull request by MikelSage**
Adds admin dashboard functionality for orders

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -22,4 +22,12 @@ def create
 
     redirect_to orders_path
   end
+
+  def update
+    order = Order.find(params[:id])
+
+    order.update(status: params[:status])
+
+    redirect_to request.referrer
```
Comment (swdonovan) [inline app/controllers/orders_controller.rb:31]: The name of this method kills me for some reason still.  Way to find that!

File: app/models/order.rb [hunk 2]
```diff
@@ -11,4 +11,21 @@ def total_price
...
+    if sort_by == 'ordered'
+      Order.ordered
+    elsif sort_by == 'paid'
+      Order.paid
+    elsif sort_by == 'cancelled'
+      Order.cancelled
+    elsif sort_by == 'completed'
+      Order.completed
+    else
+      Order.all
```
Comment (swdonovan) [inline app/models/order.rb]: Whoa I think theres a Mod 1 Repo somewhere that looks like this....but refactoring it is hard so we will have to get crafty tomorrow

**Review by swdonovan — APPROVED** [review]
Nice work dude that whole thing looked like a beast to complete! The testing alone was crazy so hell of a job while mentoring on top of that....that is baller @MikelSage

File: app/models/order.rb (hunk 2 above)
Comment (MikelSage) [reply app/models/order.rb]: Yeah, I'm thinking I could probably reduce this whole thing down to a Order.where statement, but I gotta get this lightning talk down first.

**Comment by MikelSage** [comment]
Refactored some more, made sure everything works and took out the debugging statements.

File: app/models/order.rb [hunk 3]
```diff
@@ -15,17 +15,7 @@ def self.count_by_status
...
-      Order.ordered
-    elsif sort_by == 'paid'
-      Order.paid
-    elsif sort_by == 'cancelled'
-      Order.cancelled
-    elsif sort_by == 'completed'
-      Order.completed
-    else
-      Order.all
-    end
```
Comment (swdonovan) [inline app/models/order.rb]: this is sheer proof that delirium does set in to the immortals.  skynet has won......Nice refactor

**Comment by jimszalew** [comment]
Fixed the merge conflict in routes. Everything else looks scoot-tastic!

**Pull request merged**